Lib/multiprocessing/pool.py

```diff
@@ -256,11 +282,21 @@ def _repopulate_pool(self):
             util.debug('added worker')
 
     def _maintain_pool(self):
-        """Clean up any exited workers and start replacements for them.
-        """
-        if self._join_exited_workers():
+        """Clean up any exited workers and start replacements for them."""
+        need_repopulate = self._join_exited_workers()
+        if need_repopulate:
             self._repopulate_pool()
 
+        if need_repopulate is None:
+            with self._worker_state_lock:
+                self._worker_handler._state = BROKEN
+
+            err = BrokenProcessPool(
+                'A worker in the pool terminated abruptly.')
+            # Exhaust MapResult with errors
+            for i, cache_ent in list(self._cache.items()):
```

Comment:
Out of curiosity, is there any reason why we iterate on a list of of `self._cache`?